lib/liquid/tag.rb

```diff
@@ -38,6 +38,22 @@ def render(_context)
       ''
     end
 
+    def disabled?(context, output)
+      if context.registers['disabled_tags']&.disabled?(tag_name)
+        output << disabled_response
+      end
+    end
+
+    def disabled_response
+      "#{tag_name} #{options[:locale].t('errors.disabled.tag')}"
+    end
+
+    def disable_tags(context, tags)
+      context.registers['disabled_tags'].disable(tags) do
+        yield
+      end
```

Comment:
Since we discussed this, just so you know this could also be written like this:
```ruby
def disable_tags(context, tags, &block)
  context.registers['disabled_tags'].disable(tags  &block)
end
```